Introduce the map layout item toolbar (#4573)
qgis/QGIS-Documentation: source/docs/user_manual/print_composer/composer_items/composer_map.rst

   Map Item Properties Panel

The Toolbar
-----------

> The Map :guilabel:`Item Properties` panel embeds a toolbar with following

Reviewer: ```suggestion
The Map :guilabel:`Item Properties` panel embeds a toolbar with the following
```

> functionalities:

Reviewer: ```suggestion
possibilities / functions:
```

> * |draw| :sup:`Update map preview`

Reviewer 2: ```suggestion
* |draw| :sup:`Refresh map preview`
```

Author: The text has changed in the GUI? I don't have a recent LTR build right now and it shows "Update"

Reviewer: LTR (3.4.14) has a big button labelled "Update Preview" as the first element of "Main Properties" (followed by "Scale").
3.10.0 has a small refresh-button (blue arrows biting each other's  tails) in a top "toolbar" above "Main Properties"  - the HOVER text for the button is "Update Map Preview".

* |setToCanvasExtent| :sup:`Set map canvas to match main canvas extent`
* |viewExtentInCanvas| :sup:`View current map extent in main canvas`
* |setToCanvasScale| :sup:`Set map scale to match main canvas scale`
* |viewScaleInCanvas| :sup:`Set main canvas to match current map scale`

> * |showBookmarks| :sup:`Bookmarks` to assign a bookmark to the map item

Reviewer: ```suggestion
* |showBookmarks| :sup:`Bookmarks`: assign a bookmark to the map item
```

Reviewer 2: I think "assign" is misleading here -- it could be intrepreted as saving a new bookmark from the item. Maybe "to set the map item extent for match an existing spatial bookmark":

> * |moveItemContent| :sup:`Interactively edit map extent`: allows to pan

Reviewer: ```suggestion
* |moveItemContent| :sup:`Interactively edit map extent`: pan
```

Reviewer 2: or "pan and zoom interactively"

Reviewer: @nyalldawson, how does this allow zooming.  Panning can be done by clicking in the map and dragging. The mouse wheel zooms the layout  (not the map) - same behaviour as when in "Move item content mode" (the icon seems to be a smaller version of that one).

Reviewer 2: You have to have the map selected first before the zoom scroll wheel will work

Reviewer: Yep! That works.

  within the map item

> * |labeling| :sup:`Labeling settings`: controls features labels behaviour

Reviewer: ```suggestion
* |labeling| :sup:`Labeling settings`: controls feature label behaviour
```

  (placement, visibility...) in the layout map item extent:

>   * set a :guilabel:`Margin from map edges`, a variable distance from the

Reviewer: ```suggestion
  * set a :guilabel:`Margin from map edges`, a distance from the
```

Author: this is data definable...

Reviewer: I have modified the suggestion.

    map item's limits inside which no label should be displayed
  * |unchecked| :guilabel:`Allow truncated labels on edges of map`: controls

>     whether labels which fall partially outside of the map item allowed extent

Reviewer: ```suggestion
    whether labels which fall partially outside of the map item's extent
```

Author: the map extent is not necessarily the one used; if you add margin, I assume the area used for placement checks is reduced accordingly.

Reviewer: My suggestion is only on the language. If more explanation is needed, please add it. I have tested this, and my suggestion is consistent with what I experienced when testing.

>     should be rendered. If checked, these labels will be shown (when there’s

Reviewer: ```suggestion
    should be rendered. If checked, these labels will be shown (even when there’s
```

    no way to place them fully within the visible area). If unchecked then
    partially visible labels will be skipped.
  * :guilabel:`Label blocking items`: allows other layout items (such as
    scalebars, north arrows, inset maps, etc) to be marked as a blockers for
    the map labels in the **active** map item. This prevents any map labels
    from being placed under those items - causing the labeling engine to either

>     try alternative placement for these labels (or discarding them altogether).

Reviewer: ```suggestion
    try alternative placement for these labels or discarding them altogether.
```

Reviewer: An alternative to removing the parenthesis is to skip "either".

    If a :guilabel:`Margin from map edges` is set, the map labels are not

>     placed closer than the specified distance from the checked layout items.

Reviewer: ```suggestion
    placed closer than the specified distance from the map edges.
```

Reviewer: ?

Author: > ?

NO, another thing.

Reviewer: OK. I have not tested this.

>   * :guilabel:`Show unplaced labels`: independant from the :ref:`main canvas

Reviewer: ```suggestion
  * :guilabel:`Show unplaced labels`: can be used to determine whether labels
```

>     setting <automated_placement>`, it can be used to determine whether labels

Reviewer: ```suggestion
```

>     are missing from the layout map.

Reviewer: ```suggestion
    are missing from the layout map (independent of the
    :ref:`main canvas setting <automated_placement>`).
    
```

>     are missing from the layout map.

Reviewer 2: ```suggestion
    are missing from the layout map (e.g. due to conflicts with other map labels or due to insufficient space to place the label) by highlighting them in a red color.
```

Reviewer: Fixed line length (avoiding too long lines)

Author: Updated since this setting is not really independent of the one in placement engine: they share the same color setting which may not necessarily be red

   Map Item Properties Panel

The Toolbar
-----------

The Map :guilabel:`Item Properties` panel embeds a toolbar with the following
functionalities:

* |draw| :sup:`Update map preview`
* |setToCanvasExtent| :sup:`Set map canvas to match main canvas extent`
* |viewExtentInCanvas| :sup:`View current map extent in main canvas`
* |setToCanvasScale| :sup:`Set map scale to match main canvas scale`
* |viewScaleInCanvas| :sup:`Set main canvas to match current map scale`
* |showBookmarks| :sup:`Bookmarks`: set the map item extent to match
  an existing spatial bookmark
* |moveItemContent| :sup:`Interactively edit map extent`: pan and
  zoom interactively
  within the map item
* |labeling| :sup:`Labeling settings`: control feature label behaviour
  (placement, visibility...) in the layout map item extent:

  * set a :guilabel:`Margin from map edges`, a data definable distance from the
    map item's limits inside which no label should be displayed

>   * |unchecked| :guilabel:`Allow truncated labels on edges of map`: controls

Author: Still unsatisfied with this. @nyalldawson I'm not sure the description reflects this option behavior and would appreciate your help in understanding it (unless there's a bug):
1. It seems that it disables the margin from edges (see the `YUKON-KOYUKUK` label on the right)
1. If a label is placed in the map item, should it not be kept at its position (see the `BETHEL` label at the bottom). I expected that "truncatable" labels that were not shown, if any, would be shown truncated but the ones well placed and fully displaced would keep their position? 
![truncatedisablemargin](https://user-images.githubusercontent.com/7983394/70648298-147bbd00-1c4b-11ea-98fe-17da7df1cc5d.gif)

Author: I opened a report at https://github.com/qgis/QGIS/issues/33403. Merging as is.
